py/visdom/__init__.py

```diff
@@ -2357,3 +2357,89 @@ def dual_axis_lines(self, X=None, Y1=None, Y2=None, opts=None, win=None, env=Non
             'opts': opts,
         }
         return self._send(datasend, 'events')
+
+
+    @pytorch_wrap
+    def graph(self, edges, edgeLabels = None, nodeLabels = None, opts=dict(), env=None, win=None):
+        """
+        This function draws interactive network graphs. It takes list of edges as one of the arguments.
+        The user can also provide custom edge Labels and node Labels in edgeLabels and nodeLabels respectively.
+        Along with that we have different parameters in opts for making it more user friendly. 
+
+        Args:
+            edges : list, required
+                A list of graph edges in one of the following formats (source, destination)
+            edgeLabels : list, optional
+                list of custom edge-labels. length should be equal to that of "edges"
+            nodeLabels : list, optional
+                list of custom node-labels. length should be equal to number of nodes and sequence must be in ascending order.
+            opts : dict, optional
+                * `directed` : directed (True) or undirected (False) graph; False by default
+                * `showVertexLabels` : boolean , if True displays vertex labels else hides the label; "True" by default
+                * `showEdgeLabels` :  boolean , if True displays edge labels else hides the label; "False" by default
+                * `scheme` : {"same", "different"} nodes with "same" or "diffent" colors; "same" by default
+                * `height` : height of the Pane
+                * `width` : width of the Pane
+        """
+        try:
+            import networkx as nx
+        except:
+            raise RuntimeError(
+                "networkx must be installed to plot Graph figures")
+        
+        G = nx.Graph()
+        G.add_edges_from(edges)
+        node_data = list(G.nodes())
+        link_data = list(G.edges())
```

Comment:
That's a bit awkward: you require "edges" as the input, then create a graph from it to finally extract the edges back again.
If you stick to networkx, wouldn't make it convenient to require the graph itself as the input? I mean, you already require the networkx to be installed.

Reply:
> I'm fine with the overall implementation. Accepted.
> 
> The only question is: you're trying to add a file with 9000+ lines. Do you use `package-lock.json` somewhere? How large the file is in MB? Maintainers, is there a limit on additional files to be included in the package?

Hi @dizcza. I have deleted all the extra files in my latest commit. Now everything is fine. I have tested it in a new virtual env.  and for the second part, the main thought behind this is that I didn't want the user to download any extra library besides visdom. The reason for using networkx again is only to get nodes out of edges ( which i could have done using set and for loop), and also i was checking if there is a mech inside networkx to check for errors when given wrong input this way i wanted to reduce the work for validation.

Reply:
That's fine. You or me, or anyone else could improve this in the future once we understand how to do it "better".